android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -93,19 +95,46 @@
   private MdnsUrlDiscoverer mMdnsUrlDiscoverer;
   private SsdpUrlDiscoverer mSsdpUrlDiscoverer;
   private boolean mDebugViewEnabled = false;
-  // Run when the SCAN_TIME_MILLIS has elapsed.
-  private Runnable mScanTimeout = new Runnable() {
+  private boolean mSecondScanComplete;
+
+  // Run when the FIRST_SCAN_MILLIS has elapsed.
+  private Runnable mFirstScanTimeout = new Runnable() {
     @Override
     public void run() {
-      mScanningAnimationDrawable.stop();
-      scanLeDevice(false);
-      mMdnsUrlDiscoverer.stopScanning();
-      mSsdpUrlDiscoverer.stopScanning();
-      mNearbyDeviceAdapter.sortUrls();
-      mNearbyDeviceAdapter.notifyDataSetChanged();
-      fadeInListView();
+      Log.d(TAG, "running first scan timeout");
+      if (!mPwoMetadataQueue.isEmpty()) {
```

Comment:
Ok. I guess my only concern was that the user might think scanning is done, while scanning is actually occurring. The only solution I can think of is that we keep the refresh animation going until the third timeout, but then it's blocking the top of the list view, which seems suboptimal. I guess as long as we have parity with iOS we should be ok  :)